Jenkinsfile

```diff
@@ -618,6 +620,37 @@ def dotnet(Closure body){
   }
 }
 
+def testTools(Closure body){
+  def homePath = "${env.WORKSPACE}/${env.BASE_DIR}"
+  withEnv([
+    "PATH=${homePath}/azure-functions-cli:${PATH}"
+    ]){
+    sh(label: 'Install Azure Functions Core Tools', script: """
+      # See: https://github.com/Azure/azure-functions-core-tools#other-linux-distributions
+
+    # Get the URL for the latest v4 linux-64 artifact
+    latest_v4_release_url=\$(curl -s https://api.github.com/repos/Azure/azure-functions-core-tools/releases \
+      | jq -r '.[].assets[].browser_download_url' \
+      | grep 'Azure.Functions.Cli.linux-x64.4.*zip\$' \
+      | head -n 1)
+    
+    # Preserve only the filename component of the URL
+    latest_v4_release_file=\${latest_v4_release_url##*/}
+
+    # Download the artifact
+    curl -sLO "\${latest_v4_release_url}"
+
+    # Unzip the artifact to ./azure-functions-cli
+    unzip -d azure-functions-cli "\${latest_v4_release_file}"
+
+    # Make required executables ... executable.
+    chmod +x ./azure-functions-cli/func
+    chmod +x ./azure-functions-cli/gozip
+    """)
+    body()
+  }
+}
+
```

Comment:
I considered both approaches and decided on the `Jenkinsfile` "inline version" because:
* It follows the example of the `dotnet` block we already use here:
* The `PATH` modification needs to happen in `withEnv` either way and by having everything in this block it's more discoverable what actually happens.
* The code can be pulled out almost verbatim into a script should we ever have the need.
Let me know what you think.